### PR TITLE
Changes needed to build with MinGW. Should not interfere with Unix builds.

### PR DESCRIPTION
**Pull request by joa-quim**

### CODE REVIEW

**Review by anbj — APPROVED** [review]
What should this do? Nothing? If so - it works. I'm able to compile without problems.

**Comment by PaulWessel** [comment]
I’ll give it a try on Mac

**Comment by PaulWessel** [comment]
Word on macOS: YET: I am on the origin/for-MinGW branch for testing it and I get these warnings which is not the macro definition but usage in gravies.  We need parenthesis here to avoid confusion and what is meant:

```
1 warning generated.
[113/287] Building C object src/CMakeFiles/gmtlib.dir/grdview.c.o
/Users/pwessel/UH/RESEARCH/CVSPROJECTS/GMTdev/gmt-dev/src/grdview.c:994:22: warning: '&&' within '||' [-Wlogical-op-parentheses]
                if (Ctrl->G.n == 1 && gmt_M_file_is_image (Ctrl->G.file[0]) || Ctrl->G.is_mem_image) {
                    ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~
/Users/pwessel/UH/RESEARCH/CVSPROJECTS/GMTdev/gmt-dev/src/grdview.c:994:22: note: place parentheses around the '&&' expression to silence this warning
                if (Ctrl->G.n == 1 && gmt_M_file_is_image (Ctrl->G.file[0]) || Ctrl->G.is_mem_image) {
                    ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by PaulWessel** [comment]
It is there because it is in master too.

**Review by PaulWessel — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -532,7 +532,7 @@ if (HAVE_DLADDR AND HAVE_LIBDL)
 	target_link_libraries (gmtlib dl)
 endif (HAVE_DLADDR AND HAVE_LIBDL)
 
-if (MSVC AND FFTW3_FOUND)
+if ((MSVC OR MINGW) AND FFTW3_FOUND)
```
Comment (kmilos) [inline src/CMakeLists.txt:535]: This should've been just `if (WIN32 AND FFTW3_FOUND)` as it is platform specific, not compiler/toolchain specific...

**Pull request merged**